### PR TITLE
JS-Book-Library

### PR DESCRIPTION
**Pull request by SajjadAhmad14**
- User can add books to the library.
-  User can remove a book from the library.
-  User can change the read status of a book.

### CODE REVIEW

File: assets/script/main.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+// Book constructor
+function Book() {
+  this.title = document.getElementById('title').value;
+  this.author = document.getElementById('author').value;
+  this.pages = document.getElementById('pages').value;
+  this.read = (function () {
+    if (document.getElementById('yes').checked) return document.getElementById('yes').value;
+    return document.getElementById('no').value;
+  }());
+}
```
Comment (elmejdki) [inline assets/script/main.js]: - [ ] you should always write pure functions your constructors and methods should not rely on other variables, please use a constructor that takes parameters.

**Review by elmejdki — CHANGES_REQUESTED** [review]
# status: require changes :recycle: 
Great job guys, you have implemented all the necessary functionalities :1st_place_medal:  but there is some changes :hammer: :hammer_and_wrench: :toolbox: that you need to make in order to get this project approved :checkered_flag:

please check the comments below:
:point_down: :point_down: :point_down: :point_down: :point_down:

## Required Changes :red_circle::

- [ ] Fix the live demo link you used localhost :laughing:.

- [ ] it will be better if you made the Add book button toggle the form I mean, display it and hide it on clicks.
image

- [ ] Keep the application logic separated from DOM manipulation tasks.
image

## Joyti Only Requirements :yum::
- [ ] Joyti I highly recommend that you create a Linkedin profile(a Twitter also if you can) now before tomorrow and start networking and connecting to new people cause it will be very helpful in your job search later.

## OPTIONAL:

- [ ] use localStorage to make the list of books persistent on page loads or much better try to use firebase it's really easy just give it a try.

### Please make all the required changes before you submit another code review.
## Happy Coding :tophat: 
_Feel free to contact me on slack if you have any questions or need help_
## _Zakariae El Mejdki_

File: assets/script/main.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  row.innerHTML = `
+    <td class="mx-4">${book.title}</td>
+    <td class="mx-4">${book.author}</td>
+    <td class="mx-4">${book.pages}</td>
+    <td><a href="#" class="btn btn-primary mx-4" id="statusYes">${book.read}</a></td>
+    <td><a href="#" class="btn btn-danger mx-4 delete">Remove</a></td>
+    `;
+  list.appendChild(row);
+}
```
Comment (elmejdki) [inline assets/script/main.js]: - display book function should loop through all the books for the librarybooks variable and display them one by one.
image

**Review by elmejdki — CHANGES_REQUESTED** [review]

**Review by Forison — CHANGES_REQUESTED** [review]
## Status::Change required❌
# 🙋‍♂️ team
### Good job on addressing some of the issues raised in the previous review🤗.
* [ ] I could only add a book. After I tried adding more books after the first, it does not happen. The application freezes. Let us address this issue to have the page working.
#### That is it kindly address the change and submit a re-review🤗
#### Happy coding

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon:

Hi **TEAM** congratulations on this project :fireworks: Having fixed all the comments outlined by he previous TSE's :smile: I hereby approve the project :+1: 

Kindly merge into the development, do not forget to submit the project completions form
_Happy coding: :man_technologist: :woman_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**